### ISSUE TITLE
Improve Screen Reader Accessibility for Interactive Mode

### ISSUE THREAD

**Issue by vick08** [anthropics/claude-code#5586]
**Bug Description**
Currently, it is difficult to use Claude Code interactive mode with screen readers, e.g. VoiceOver for Mac OS. Major issues include:
1. The presence of decorative characters that frame the TUI produce a lot of extraneous speech when the user attempts to read the output.
2. Because of the lack of a physical cursor or a highlight screen readers have difficulty tracking the current line and what the user is typing. FOr example, there is no echo as the characters or words are typed into the prompt field.
Three are other issues, but I think the above ones are top priority..
Please consider accessibility if possible. I realize the "ink" rendering engine may be a cause here, but I would like to encourage to explore some sort of a "screen reader" mode where accessibility enhancements could be introduced. Thank you for reading and considering!

**Environment Info**
- Platform: darwin
- Terminal: Apple_Terminal
- Version: 1.0.73
- Feedback ID: e2687002-22c6-404e-99da-b6dfb51e3786

**Errors**
```json
[]
```
